src/ECharts/Axis.purs

```diff
@@ -0,0 +1,373 @@
+    "formatter" := a.formatter,
+    "textStyle" := a.textStyle,
+    "interval" := a.interval,
+    "rotate" := a.rotate,
+    "margin" := a.margin,
+    "clickable" := a.clickable
+    ]
+axisLabelDefault = {
+  show: Nothing,
+  formatter: Nothing,
+  textStyle: Nothing,
+  interval: Nothing,
+  rotate: Nothing,
+  margin: Nothing,
+  clickable: Nothing
+  }
+
+
+data Axises = OneAxis Axis | TwoAxises Axis Axis
+instance axisesEncodeJson :: EncodeJson Axises where
+  encodeJson (OneAxis axis) = encodeJson axis
+  encodeJson (TwoAxises axis axis2) = encodeJson [axis, axis2]
+
+
+newtype AxisSplitLine =
+  AxisSplitLine {
+    "show" :: Maybe Boolean,
+    "onGap" :: Maybe Boolean,
+    "lineStyle" :: Maybe LineStyle
+    }
+instance axisSplitLineEncodeJson :: EncodeJson AxisSplitLine where
+  encodeJson (AxisSplitLine obj) =
+    fromObject $ fromList $
+    [
+      "show" := obj.show,
+      "onGap" := obj.onGap,
+      "lineStyle" := obj.lineStyle
+    ]
+axisSplitLineDefault = {
+  show: Nothing,
+  onGap: Nothing,
+  lineStyle: Nothing
+  }
+
+
+newtype AxisSplitArea =
+  AxisSplitArea {
+    "show" :: Maybe Boolean,
+    "onGap" :: Maybe Boolean,
+    "areaStyle" :: Maybe AreaStyle
+    }
+instance axisSplitAreaEncodeJson :: EncodeJson AxisSplitArea where
+  encodeJson (AxisSplitArea obj) =
+    fromObject $ fromList $
+    [
+      "show" := obj.show,
+      "onGap" := obj.onGap,
+      "areaStyle" := obj.areaStyle
+    ]
+axisSplitAreaDefault = {
+  show: Nothing,
+  onGap: Nothing,
+  areaStyle: Nothing
+  }
+
+  
+data AxisType = CategoryAxis | ValueAxis | TimeAxis
+instance axisTypeEncodeJson :: EncodeJson AxisType where
+  encodeJson a = encodeJson $ case a of
+    CategoryAxis -> "category"
+    ValueAxis -> "value"
+    TimeAxis -> "time"
+
+
+data AxisPosition = LeftAxis | RightAxis | TopAxis | BottomAxis
+instance axisPositionEncodeJson :: EncodeJson AxisPosition where
+  encodeJson a = encodeJson $ case a of
+    LeftAxis -> "left"
+    RightAxis -> "right"
+    TopAxis -> "top"
+    BottomAxis -> "bottom"
+
+
+data AxisNameLocation = Start | End
+instance axisNameLocationEncodeJson :: EncodeJson AxisNameLocation where
+  encodeJson a = encodeJson $ case a of
+    Start -> "start"
+    End -> "end"
+
+
+
+data AxisData = CommonAxisData String
+              | CustomAxisData {"value" :: String, "textStyle" :: TextStyle}
+instance axisDataEncodeJson :: EncodeJson AxisData where
+  encodeJson (CommonAxisData name) = fromString name
+  encodeJson (CustomAxisData obj) =
+    fromObject $ fromList $
+    [
+      "value" := obj.value,
+      "textStyle" := obj.textStyle
+    ]
+
+data AxisBoundaryGap = CatBoundaryGap Boolean
+                   | ValueBoundaryGap Number Number
+
+instance axisBoundaryGapEncodeJson :: EncodeJson AxisBoundaryGap where
+  encodeJson (CatBoundaryGap bool) = encodeJson bool
+  encodeJson (ValueBoundaryGap num1 num2) = encodeJson [num1, num2]
+
+
+newtype Axis =
+  Axis {
+    "type" :: Maybe AxisType,
```

Comment:
These don't have to be quoted -- the whole library should be consistent and not quote any fields either in types or record literals (anywhere it doesn't have to).